bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -66,6 +67,21 @@ def __init__(self, filename, fileindex):
         self.filename = filename
         self.fileindex = fileindex
 
+    def __lt__(self, other):
+        """Total ordering is defined to be able to sort a set[FileAndIndex]."""
+        return self.fileindex < other.fileindex 
+
+    def __eq__(self, other):
+        """Defined in compliance with total_ordering decorator"""
+        return self.fileindex == other.fileindex
+
+    def __hash__(self):
+        """
+        This type needs to be hashable to be part of a set in
+        get_clustered_ranges
+        """
+        return hash(self.filename)
```

Comment:
could this make it harder to run twice on the same file?

Reply:
In general I wouldn't think so since the hashing is just to be able to put objects of this class in a Python `set`. The info that allows running twice on the same file is just the `fileindex` in the list of input files.

That being said, as soon as we switch to using `TEntryList` we won't be able to support running on the same file twice due to  https://github.com/root-project/root/issues/8505

Reply:
Anyway, if we want to be on the safe side, I could just make this `hash(self.filename) + hash(self.fileindex)`

Reply:
ah no ok, it's fine: `FileAndIndex` objects with same fileindex will also always have same filename